### ISSUE TITLE
[Card] Real World Examples have invisible focus indicators in Firefox and Safari

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3506]
### 😯 Current Behavior

Focus indicator is invisible when placed on whole cards. It is visible only on single buttons inside cards but when it is placed on whole card link it's not visible. 

This happens only in Firefox and Safari.

https://github.com/user-attachments/assets/3e85081d-029e-45cc-99b4-6aa5e907b003

### 🤔 Expected Behavior/possible solution?

Focus on whole cards should be visible like it is on Chrome.

https://github.com/user-attachments/assets/6f736b4c-e68a-4afe-92ac-176ce74a72ce

### 🖥️ Steps to Reproduce

1. In Firefox and Safari go to https://storybook.sanomalearning.design/?path=/story/layout-card--real-world-examples
2. With tab set keyboard focus on each card link
3. Check if focus indicator is visible on each card link

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/layout-card--real-world-examples

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Firefox, Safari

### Component version

2.0.2

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
